pkg/controller/plan/adapter/ova/builder.go

```diff
@@ -0,0 +1,594 @@
+			})
+	} else {
+		for _, disk := range vm.Disks {
+			env = append(env,
+				core.EnvVar{
+					Name:  "V2V_diskPath",
+					Value: getDiskSourcePath(vm.OvaPath, disk.Name),
+				})
+		}
+		env = append(env,
+			core.EnvVar{
+				Name:  "V2V_provider",
+				Value: "ovf",
+			})
+	}
+	return
+}
+
+// Build the DataVolume credential secret.
+func (r *Builder) Secret(vmRef ref.Ref, in, object *core.Secret) (err error) {
+	// TODO only if we want to save some data for DV.
+	return
+}
+
+// Create DataVolume specs for the VM.
+func (r *Builder) DataVolumes(vmRef ref.Ref, secret *core.Secret, _ *core.ConfigMap, dvTemplate *cdi.DataVolume) (dvs []cdi.DataVolume, err error) {
+	vm := &model.VM{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+		return
+	}
+
+	dsMapIn := r.Context.Map.Storage.Spec.Map
+	for i := range dsMapIn {
+		mapped := &dsMapIn[i]
+		ref := mapped.Source
+		ds := &model.Disk{}
+		fErr := r.Source.Inventory.Find(ds, ref)
+		if fErr != nil {
+			err = fErr
+			return
+		}
+		for _, disk := range vm.Disks {
+			if disk.ID == ds.ID {
+				diskSize, err := getResourceCapacity(disk.Capacity, disk.CapacityAllocationUnits)
+				if err != nil {
+					return nil, err
+				}
+				storageClass := mapped.Destination.StorageClass
+				var dvSource cdi.DataVolumeSource
+				// Let virt-v2v do the copying
+				dvSource = cdi.DataVolumeSource{
+					Blank: &cdi.DataVolumeBlankImage{},
+				}
+				dvSpec := cdi.DataVolumeSpec{
+					Source: &dvSource,
+					Storage: &cdi.StorageSpec{
+						Resources: core.ResourceRequirements{
+							Requests: core.ResourceList{
+								core.ResourceStorage: *resource.NewQuantity(diskSize, resource.BinarySI),
+							},
+						},
+						StorageClassName: &storageClass,
+					},
+				}
+				// set the access mode and volume mode if they were specified in the storage map.
+				// otherwise, let the storage profile decide the default values.
+				if mapped.Destination.AccessMode != "" {
+					dvSpec.Storage.AccessModes = []core.PersistentVolumeAccessMode{mapped.Destination.AccessMode}
+				}
+				if mapped.Destination.VolumeMode != "" {
+					dvSpec.Storage.VolumeMode = &mapped.Destination.VolumeMode
+				}
+
+				dv := dvTemplate.DeepCopy()
+				dv.Spec = dvSpec
+				if dv.ObjectMeta.Annotations == nil {
+					dv.ObjectMeta.Annotations = make(map[string]string)
+				}
+				dv.ObjectMeta.Annotations[planbase.AnnDiskSource] = getDiskSourcePath(disk.FilePath, disk.Name)
+				dvs = append(dvs, *dv)
+			}
+		}
+	}
+
+	return
+}
+
+// Create the destination Kubevirt VM.
+func (r *Builder) VirtualMachine(vmRef ref.Ref, object *cnv.VirtualMachineSpec, persistentVolumeClaims []core.PersistentVolumeClaim) (err error) {
+	vm := &model.VM{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+		return
+	}
+
+	if r.Plan.Spec.Warm {
```

Comment:
this is too late to discover this - we should validate it before

Reply:
@liranr23 doesn't it already verified as part of the migration steps in kubevirt.go? do we need to add a separate validation for OVA?

Reply:
the validator can do the validation if we didn't add it there already (provider == ova and warm).

Reply:
looks like we are doing it in the plan validation (`pkg/controller/plan/adapter/ova/validator.go`):
```// Validate whether warm migration is supported from this provider type.
func (r *Validator) WarmMigration() (ok bool) {
	ok = false
	return
}
```
so we don't need it in the builder.